lib/parsers/webp_parser.rb

```diff
@@ -132,27 +127,30 @@ def augment_image(image)
     xmp = nil
     num_frames = 0
     loop do
-      chunk_code, chunk_size = safe_read(@buf, 8).unpack('A4V')
+      fourCC, chunk_size = safe_read(@buf, 8).unpack('A4V')
       safe_skip(@buf, 1) if chunk_size.odd? # Padding byte of 0 added if chunk size is odd.
-      case chunk_code
+      case fourCC
       when 'EXIF'
         exif ||= exif_from_tiff_io(StringIO.new(safe_read(@buf, chunk_size)))
       when 'XMP'
-        xmp ||= parse_xmp_metadata(chunk_size)
+        safe_skip(@buf, chunk_size)
       when 'ANMF'
         num_frames += 1 if has_multiple_frames
         safe_skip(@buf, chunk_size)
       else
         safe_skip(@buf, chunk_size)
       end
     end rescue FormatParser::IOUtils::InvalidRead
-    image.height_px, image.width_px = image.width_px, image.height_px if exif&.rotated?
+
     if exif || xmp
       image.intrinsics = {}
-      image.intrinsics[:exif] = exif if exif
+      if exif
+        image.height_px, image.width_px = image.width_px, image.height_px if exif.rotated?
+        image.intrinsics[:exif] = exif
+        image.orientation = exif.orientation_sym
+      end
       image.intrinsics[:xmp] = xmp if xmp
     end
```

Comment:
If we're cool with `intrinsics` being an empty hash instead of `nil`, then it becomes trivial:
```
image.intrinsics = {}
if exif
  image.height_px, image.width_px = image.width_px, image.height_px if exif.rotated?
  image.intrinsics[:exif] = exif
  image.orientation = exif.orientation_sym
end
image.intrinsics[:xmp] = xmp if xmp
```
but that may well not be the case.

Reply:
thought EXIF and XMP where mutually exclusive! But if you say they can be both be there then sure, definitely fine with it! Only thing about the last option: check out the other formats, just to make sure to be consistent. If other formats do also empty hash, go for it. If instead they all do `nil` then let's keep it consistent

Reply:
Yeah I thought they might be too, but after checking some of my own photos, they're definitely both there! Awesome, I'll check. Thanks!